sympy/codegen/ast.py

```diff
@@ -1468,7 +1469,12 @@ def as_Declaration(self, **kwargs):
         >>> from sympy.codegen.ast import Variable
         >>> x = Variable('x')
         >>> decl1 = x.as_Declaration()
-        >>> decl1.variable.value == None
+        # value is special NoneToken() which must be tested with == operator
+        >>> decl1.variable.value is None  # won't work
+        False
+        >>> decl1.variable.value == None  # not PEP-8 compliant
+        True
+        >>> decl1.variable.value == NoneToken()  # OK
```

Comment:
I think these notes should go in the NoneToken docstring only. The other docstrings should just do `== NoneToken()`.

Reply:
Also `== None` is fine IMO.

Reply:
Hello! Thank you for review.

Here are my reasonings:
Using `== None` is not consistent with Python guidelines. And what's the purpose of `NoneToken` would be then?

I agree that latest comments should go to the `NoneToken` docstring.

The whole `ast` module is internal for the developers of sympy, and I wouldn't touch it (because you guys know it better anyway) for any other reason than a bit of consistency, since I've already 'grep'ed across all the codebase.

Reply:
`NoneToken` is a symbolic representation of `None` for certain code generation contexts. We cannot use `None` directly because expressions in SymPy need to be `Basic` subclasses. I'm actually not a huge fan of it, but I think we couldn't think of a way to do certain things. 

sympy.codegen.ast isn't an internal module. It is intended to be used by end-users.